services/datamanager/builtin/file_deletion_test.go

```diff
@@ -1,332 +0,0 @@
-			fileList:                []string{"0.prog", "1.prog", "2.prog", "3.prog", "4.prog", "5.prog"},
-			expectedDeleteFilenames: []string{},
-		},
-		{
-			name:                    "file deleter should not delete non datacapture files",
-			fileList:                []string{"0.fe", "1.fi", "2.fo", "3.fum", "4.foo", "5.capture"},
-			expectedDeleteFilenames: []string{"5.capture"},
-		},
-		{
-			name:                "if cancelled context is cancelled, file deleter should return an error",
-			shouldCancelContext: true,
-		},
-	}
-
-	for _, tc := range tests {
-		t.Run(tc.name, func(t *testing.T) {
-			tempCaptureDir := t.TempDir()
-			logger := logging.NewTestLogger(t)
-			mockClient := mockDataSyncServiceClient{
-				succesfulDCRequests: make(chan *v1.DataCaptureUploadRequest, 100),
-				failedDCRequests:    make(chan *v1.DataCaptureUploadRequest, 100),
-				fail:                &atomic.Bool{},
-			}
-
-			var syncer datasync.Manager
-			if tc.syncEnabled {
-				filesToSync := make(chan string)
-				defer close(filesToSync)
-				syncer = datasync.NewManager("rick astley", mockClient, logger, tempCaptureDir, datasync.MaxParallelSyncRoutines, filesToSync)
-				defer syncer.Close()
-			}
-
-			filepaths := writeFiles(t, tempCaptureDir, tc.fileList)
-			for _, file := range tc.syncerInProgressFiles {
-				syncer.MarkInProgress(filepaths[file])
-			}
-
-			ctx, cancelFunc := context.WithCancel(context.Background())
-			defer cancelFunc()
-			if tc.shouldCancelContext {
-				cancelFunc()
-			}
-			deletedFileCount, err := deleteFiles(ctx, syncer, defaultDeleteEveryNth, tempCaptureDir, logger)
-			if tc.shouldCancelContext {
-				test.That(t, err, test.ShouldBeError, context.Canceled)
-			} else {
-				test.That(t, err, test.ShouldBeNil)
-				test.That(t, deletedFileCount, test.ShouldEqual, len(tc.expectedDeleteFilenames))
-				// get list of all files still in capture dir after deletion
-				files := getFiles(t, tempCaptureDir)
-				for _, deletedFile := range tc.expectedDeleteFilenames {
-					test.That(t, files, test.ShouldNotContain, deletedFile)
-				}
-			}
-		})
-	}
-}
-
-func writeFiles(t *testing.T, dir string, filenames []string) map[string]string {
-	t.Helper()
-	fileContents := []byte("never gonna let you down")
-	filePaths := map[string]string{}
-	for _, filename := range filenames {
-		filePath := fmt.Sprintf("%s/%s", dir, filename)
-		err := os.WriteFile(filePath, fileContents, 0o755)
-		test.That(t, err, test.ShouldBeNil)
-		filePaths[filename] = filePath
-	}
-	return filePaths
-}
-
-func getFiles(t *testing.T, path string) []string {
-	t.Helper()
-	dir, err := os.Open(path)
-	test.That(t, err, test.ShouldBeNil)
-	defer dir.Close()
-	files, err := dir.Readdir(-1)
-	test.That(t, err, test.ShouldBeNil)
-	output := []string{}
-	for _, file := range files {
-		output = append(output, file.Name())
-	}
-	return output
-}
-
-func TestFilePolling(t *testing.T) {
```

Comment:
this test was moved here: https://github.com/viamrobotics/rdk/pull/4161/files#diff-b14b206ee607e2cb14407d87e6962f7a4ab250faeddd904fc6d0c67d78559a5cR192